Get richnav.yml building again

dougbu opened the pull request:
- correct build.cmd location
- generally align richnav.yml with ci.yml
  - ignore signing, internal download args, and publish args

nit: correct name of one build step in ci.yml

dougbu committed:
Get richnav.yml building again
- correct build.cmd location
- generally align richnav.yml with ci.yml
  - ignore signing, internal download args, and publish args

nit: correct name of one build step in ci.yml

dougbu committed:
Disable pipeline in PR validation
- also remove irrelevant trigger branches

dougbu commented on .azure/pipelines/richnav.yml:
```diff
@@ -6,58 +6,73 @@
-    - internal/release/*
+
+# Do not run this pipeline for PR validation.
+pr: none
```
@jepetty @brianaeng is there any point in running this pipeline for PRs❔

jepetty replied:
Benefit would be you could use github.dev to review PRs with Code Index enabled.

Since the repo is pretty hefty, let's start with just default branch, then can evaluate enabling PRs if more folks on the team use github.dev for PR review, or once we get native github.com integration for PRs

dougbu replied:
Thanks @jepetty. I'll leave this as-is for now unless I hear the _only_ value developers perceive is for PR navigation. "As-is" means rich nav information is uploaded for 'main' and our servicing branches.

dougbu commented:
Help please @jepetty @brianaeng: What is causing the pipeline to fail now❔ It appears the agents may be missing a component but that couldn't be the case (because other richnav pipelines build using the same agents).

``` text
D:\a\_work\1\s\.packages\richcodenav.envvardump\0.1.1832-alpha\build\RichCodeNav.EnvVarDump.targets(41,5): error MSB4062: The "CLCommandLine" task could not be loaded from the assembly C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\\Microsoft.Build.CPPTasks.Common.dll. Could not load file or assembly 'file:///C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.Build.CPPTasks.Common.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\a\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\IISLib\IISLib.vcxproj]
D:\a\_work\1\s\.packages\richcodenav.envvardump\0.1.1832-alpha\build\RichCodeNav.EnvVarDump.targets(41,5): error MSB4062: The "CLCommandLine" task could not be loaded from the assembly C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\\Microsoft.Build.CPPTasks.Common.dll. Could not load file or assembly 'file:///C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.Build.CPPTasks.Common.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\a\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\gtest\gtest.vcxproj]
D:\a\_work\1\s\.packages\richcodenav.envvardump\0.1.1832-alpha\build\RichCodeNav.EnvVarDump.targets(41,5): error MSB4062: The "CLCommandLine" task could not be loaded from the assembly C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\\Microsoft.Build.CPPTasks.Common.dll. Could not load file or assembly 'file:///C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.Build.CPPTasks.Common.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\a\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\gtest\gtest.vcxproj]
D:\a\_work\1\s\.packages\richcodenav.envvardump\0.1.1832-alpha\build\RichCodeNav.EnvVarDump.targets(41,5): error MSB4062: The "CLCommandLine" task could not be loaded from the assembly C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\\Microsoft.Build.CPPTasks.Common.dll. Could not load file or assembly 'file:///C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.Build.CPPTasks.Common.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\a\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\IISLib\IISLib.vcxproj]
D:\a\_work\1\s\.packages\richcodenav.envvardump\0.1.1832-alpha\build\RichCodeNav.EnvVarDump.targets(41,5): error MSB4062: The "CLCommandLine" task could not be loaded from the assembly C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\\Microsoft.Build.CPPTasks.Common.dll. Could not load file or assembly 'file:///C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.Build.CPPTasks.Common.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\a\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\IISLib\IISLib.vcxproj]
D:\a\_work\1\s\.packages\richcodenav.envvardump\0.1.1832-alpha\build\RichCodeNav.EnvVarDump.targets(41,5): error MSB4062: The "CLCommandLine" task could not be loaded from the assembly C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\\Microsoft.Build.CPPTasks.Common.dll. Could not load file or assembly 'file:///C:\Program Files %28x86%29\Microsoft Visual Studio\2019\BuildTools\MSBuild\Microsoft\VC\v160\Microsoft.Build.CPPTasks.Common.dll' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\a\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\gtest\gtest.vcxproj]
```

brianaeng commented:
@dougbu that's an old version of our `envvardump` package so we're thinking a package update there will fix it. I'll update it in the .NET Arcade repo, but in the meantime can you please try filtering out C++? 

I think all of our other .NET org repos are only running C# so that's why they may be succeeded when we're failing here.

dougbu commented:
@brianaeng our current language setting is `richCodeNavigationLanguage: 'csharp,typescript,java'` so I can't remove C++. I can remove Java and TypeScirpt but that implies there's a bug in interpreting the language property.

dougbu commented on .azure/pipelines/jobs/default-build.yml:
```diff
@@ -85,7 +85,7 @@ jobs:
     ${{ if ne(parameters.enableRichCodeNavigation, '') }}:
       enableRichCodeNavigation: true
-      richCodeNavigationLanguage: 'csharp,typescript,java'
+      richCodeNavigationLanguage: 'csharp'
```
This didn't change anything and https://dev.azure.com/dnceng/public/_build/results?buildId=1665654&view=results failed the same way as previous attempts. I suspect successful richnav pipelines are either all using `dotnet msbuild` exclusively or just have no C++ code in the repo. As a hack, could disable the native build entirely. Is that the right way forward❔

brianaeng replied:
It looks like you tried that but it still didn't work ☹️ Thanks for hacking away at this! We could also try filtering out the files via the trigger path fields like the runtime repo does [here](https://github.com/dotnet/runtime/commit/eb03e0f7bc396736c7ac59cf8f135d7c632860dd). We do successfully index other C++ repos though, so I'm hoping the package upgrade does the fix -- that PR is approved and awaiting merge [here](https://github.com/dotnet/arcade/pull/8619).

dougbu committed:
!fixup! Try doing native build w/o rich nav

nit: put `-no`s together

dougbu committed:
!fixup! Disable rich nav explicitly
- also remove installers build entirely; build will fail

nits:
- don't bother setting `$(EnableRichCodeNavigation)` or `$(OnlyPackPlatformSpecificPackages)` to `true`
  - job.yml sets `$(EnableRichCodeNavigation)` variable and `msbuild` will find that in environment
  - `$(OnlyPackPlatformSpecificPackages)` is not relevant in this pipeline; not packing

dougbu commented on .azure/pipelines/richnav.yml:
```diff
@@ -6,58 +6,68 @@
 
 variables:
 - name: _BuildArgs
   value: '/p:SkipTestBuild=true'
```
Thoughts on whether turning off rich navigation for test code makes sense anyone❔

dougbu commented on .azure/pipelines/richnav.yml:
```diff
@@ -6,58 +6,68 @@
                 -arch x64
-                /p:EnableRichCodeNavigation=true
+                -buildNative
+                /p:EnableRichCodeNavigation=false
```
Setting things this way overrides the environment variable created for the `$(enableRichCodeNavigation)` variable job.yml sets based on the parameter. Given https://dev.azure.com/dnceng/public/_build/results?buildId=1667542&view=results worked, this seems about right though I wonder how repos are successfully processing C++ code. Perhaps they all use `cmake` instead of `cl`❔

dougbu commented:
/ping reviewers

dougbu commented:
Pipeline YAML is specific to the branch. Backporting to release/6.0 so that we at least don't turn all PRs there yellow.

dougbu commented:
/backport to release/6.0
